sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -549,9 +581,15 @@ class StreamExecution(
           cd.dataType, cd.timeZoneId)
     }
 
+    // Reset confs to disallow change in number of partitions
```

Comment:
Why need to set the confs for every batch? You can set it after recovering `offsetSeqMetadata`.